core/src/test/java/org/bitcoinj/core/TransactionInputTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2011 Google Inc.
+ * Copyright 2014 Andreas Schildbach
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bitcoinj.core;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+
+import org.bitcoinj.params.UnitTestParams;
+import org.bitcoinj.script.ScriptBuilder;
+import org.bitcoinj.testing.FakeTxBuilder;
+import org.bitcoinj.wallet.AllowUnconfirmedCoinSelector;
+import org.bitcoinj.wallet.SendRequest;
+import org.bitcoinj.wallet.Wallet;
+import org.junit.Test;
+
+import com.google.common.collect.Lists;
+
+public class TransactionInputTest {
+
+    @Test
+    public void testStandardWalletDisconnect() throws Exception {
+        NetworkParameters params = UnitTestParams.get();
+        Wallet w = new Wallet(new Context(params));
+        w.setCoinSelector(new AllowUnconfirmedCoinSelector());
+        Address a = w.currentReceiveAddress();
+        Transaction tx1 = FakeTxBuilder.createFakeTxWithoutChangeAddress(params, Coin.COIN, a);
+        w.receivePending(tx1, null);
+        Transaction tx2 = new Transaction(params);
+        tx2.addOutput(Coin.valueOf(99000000), new ECKey());
+        w.completeTx(SendRequest.forTx(tx2));
+
+        TransactionInput txInToDisconnect = tx2.getInput(0);
+
+        assertEquals(tx1, txInToDisconnect.getOutpoint().fromTx);
+        assertNull(txInToDisconnect.getOutpoint().connectedOutput);
+        
+        txInToDisconnect.disconnect();
+
+        assertNull(txInToDisconnect.getOutpoint().fromTx);        
+        assertNull(txInToDisconnect.getOutpoint().connectedOutput);
+    }
+    
+    
```

Comment:
Minor: One newline should be enough to separate methods. Also at the end of the class no newline is needed.